### PR TITLE
Make provider run command inside of docker

### LINKED ISSUE #18: Make provider run command inside of docker
To prevent potential exploit directly to the host, all commands should be run inside of a one-time-purpose docker instance. This also helps with having one consistent docker runtime for all data providers to support similar programs with similar versions / environments.

### PR DESCRIPTION
**Pull request by taobun**
fixed #18 

- Add docker to `band-validator` image
- Provider run `docker run` instead of call directly as sub process

### CODE REVIEW

File: docker-provider/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FROM alpine:3.7
```
Comment (sorawit) [inline docker-provider/Dockerfile]: `chain/images/provider-vm/Dockerfile`?

File: chain/Dockerfile [hunk 2]
```diff
@@ -5,7 +5,7 @@ WORKDIR /zoracle
 
 RUN apt-get update && \
     apt-get install -y --no-install-recommends \
-    curl \
+    curl docker.io\
```
Comment (evilpeach) [inline chain/Dockerfile]: `curl docker.io \` add more space

**Review by sorawit — CHANGES_REQUESTED** [review]

File: docker-provider/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+
+RUN apk update && \
+    apk add curl && \
```
Comment (evilpeach) [inline docker-provider/Dockerfile]: `apk add --no-cache`

**Review by evilpeach — APPROVED** [review]

**Review by sorawit — COMMENTED** [review]
LGTM. @prin-r Please review this change.

**Review by prin-r — APPROVED** [review]

**Review by sorawit — APPROVED** [review]

**Pull request merged**